src/cmd/ksh93/bltins/print.c

```diff
@@ -298,7 +298,7 @@ int    b_print(int argc, char *argv[], Shbltin_t *context)
 		n = sh_iocheckfd(shp,fd);
 	if(!(n&IOWRITE))
 	{
-		/* don't print error message for stdout for compatibility */
+		/* for compatibility, don't print an error message to stdout */
```

Comment:
Error messages are never printed to stdout. I think the original comment is correct.